### PR TITLE
Periodical Refresh of DnsNameResolver

### PR DESCRIPTION
**Pull request by sdorazio**
This pull request is to address issue #2514. The implementation uses the cache time configured with Java's [networkaddress.cache.ttl](http://docs.oracle.com/javase/6/docs/technotes/guides/net/properties.html) to periodically refresh the resolved addresses.

### CODE REVIEW

**Comment by sdorazio** [comment]
I signed it!

File: core/src/main/java/io/grpc/internal/DnsNameResolver.java [hunk 1]
```diff
@@ -120,6 +124,13 @@ public final synchronized void start(Listener listener) {
     executor = SharedResourceHolder.get(executorResource);
     this.listener = Preconditions.checkNotNull(listener, "listener");
     resolve();
+
+    Long addressTtl = Longs.tryParse(Strings.nullToEmpty(
+          java.security.Security.getProperty("networkaddress.cache.ttl")));
```
Comment (ejona86) [inline core/src/main/java/io/grpc/internal/DnsNameResolver.java]: Hmm... so if it is the default then we don't poll? The default varies, but is 30 seconds in many cases which means that polling would be useful (although I'm not saying polling every 30 seconds is what we want).

**Comment by ejona86** [comment]
okay to test

**Comment by ejona86** [comment]
@zhangkun83, it seems this slipped through the cracks. Who should review this?

**Comment by zhangkun83** [comment]
I don't feel very confident on figuring out what the right behavior should be. Maybe you or @carl-mastrangelo ?

**Comment by ejona86** [comment]
@sdorazio, I want to mention that this is actually a cross-language issue we've had trouble coming up with a design. I'd have to look at the DNS landscape a bit more, but I do have a concern that polling could always return different addresses (which would trigger unnecessary reconnects as part of the DNS server itself providing round-robin). It seems discussion should continue in #2514 to arrive at an agreed approach before reviewing an implementation.

**Comment by ejona86** [comment]
Closing to let discussion in #2514 complete. If this is the route we want to go, we can reopen this PR.

**Pull request closed without merging**